lib/course_planner/notifier.ex

```diff
@@ -6,19 +6,35 @@ defmodule CoursePlanner.Notifier do
   alias CoursePlanner.{User, Mailer, Mailer.UserEmail}
   require Logger
 
+  defmodule Notification do
+    @moduledoc "Notification struct to be used by Notifier"
+    defstruct type: nil, resource_path: "/", to: nil
+
+    def new, do: %Notification{}
+
+    def type(%Notification{} = notification, type) when is_atom(type),
+      do: %{notification | type: type}
+
+    def resource_path(%Notification{} = notification, path) when is_binary(path),
+      do: %{notification | resource_path: path}
+
+    def to(%Notification{} = notification, %User{} = user),
+      do: %{notification | to: user}
+  end
+
   @spec start_link :: GenServer.start_link
   def start_link do
     GenServer.start_link(__MODULE__, [], name: __MODULE__)
   end
 
-  @spec notify_user(User, atom()) :: GenServer.cast
-  def notify_user(%User{} = user, notification_type) do
-    GenServer.cast(__MODULE__, {:send, user, notification_type})
+  @spec notify_user(Notification.t) :: GenServer.cast
+  def notify_user(%Notification{} = notification) do
+    GenServer.cast(__MODULE__, {:send_email, notification})
   end
 
-  @spec handle_cast({atom(), User, atom()}, any()) :: {:noreply, any()}
-  def handle_cast({:send, user, notification_type}, state) do
-    email = UserEmail.build_email(user, notification_type)
+  @spec handle_cast({atom(), Notification.t}, any()) :: {:noreply, any()}
+  def handle_cast({:send_email, %{to: user, type: type, resource_path: path}}, state) do
+    email = UserEmail.build_email(user, type, path)
```

Comment:
I left it that way to change less function signatures before feedback, so yeah we can do that as well, it would get cleaner.